server/api/db/sqldb/db.py

```diff
@@ -2692,22 +2692,48 @@ def refresh_project_summaries(
         session: Session,
         project_summaries: list[mlrun.common.schemas.ProjectSummary],
     ):
-        project_summaries_to_update = []
-        for project_summary_schema in project_summaries:
-            # To avoid a race condition with the delete project flow where a project might be deleted
-            # after its summary has been queried but before the transaction is completed, we check
-            # that the project summary is not `None` after querying.
-            # This is why we cannot query all project summaries outside the loop.
-            query = self._query(
-                session, ProjectSummary, project=project_summary_schema.name
+        """
+        This method updates the summaries of projects that have associated projects in the database
+        and removes project summaries that no longer have associated projects.
+        """
+
+        summary_dict = {summary.name: summary.dict() for summary in project_summaries}
+
+        # Create a query for project summaries with associated projects
+        existing_summaries_query = (
+            session.query(ProjectSummary)
+            .outerjoin(Project, Project.name == ProjectSummary.project)
+            .filter(ProjectSummary.project.in_(summary_dict.keys()))
+        )
+
+        associated_summaries = existing_summaries_query.filter(
+            Project.id.is_not(None)
+        ).all()
+
+        # Update the summaries of projects that have associated projects
+        for project_summary in associated_summaries:
+            project_summary.summary = summary_dict.get(project_summary.project)
+            project_summary.updated = datetime.now(timezone.utc)
+            session.add(project_summary)
+
+        # To avoid race conditions where a project might be deleted after its summary is queried
+        # but before the transaction completes, we delete project summaries that do not have
+        # any associated projects.
+        orphaned_summaries = existing_summaries_query.filter(
+            Project.id.is_(None)
```

Comment:
In the outer join we filter on `Project.name`. if there is no project, will we still get it even if we don't check for `id.is_(None)`?

Reply:
we need that. also, add a unit test to verify that the flow is working as expected.